### PR TITLE
SXP loader

### PR DESCRIPTION
**Pull request by rettigl**
PR for drafting an SXP loader

### CODE REVIEW

**Comment by rettigl** [comment]
This should more or less work now, needs to be tested with a full run. For test data upload, I am waiting for David's approval.

**Comment by rettigl** [comment]
This works for now. It has only a few modifications to the flash loader, but probably too many to become a single loader. 
The metadata stuff is untouched as of now, this needs to be updated once it's known how it works. Also, the additions from the flash consistency check can be ported once completed, but this I would do in separate PRs.

File: sed/loader/flash/utils.py
Comment (zain-sohail) [inline sed/loader/flash/utils.py:1]: I think this is acciedently deleted

File: tests/data/loader/sxp/config.yaml
Comment (zain-sohail) [inline tests/data/loader/sxp/config.yaml:48]: Not for this PR necessarily, but coming up with a schema for the dataframe in config file would be great. 
I think we had the discussion to use the group_names in mpes too.

File: tests/data/loader/sxp/RAW-R0016-DA03-S00000.h5
Comment (zain-sohail) [inline tests/data/loader/sxp/RAW-R0016-DA03-S00000.h5:1]: Is this a test file?

File: tutorial/5_sxp_workflow.ipynb
Comment (zain-sohail) [inline tutorial/5_sxp_workflow.ipynb:1]: Lets wait for data approval before merging, I'd suggest

File: sed/loader/sxp/metadata.py
Comment (zain-sohail) [inline sed/loader/sxp/metadata.py:1]: It would be better to delete this in case of SXP and just leave the metadata method empty

File: sed/loader/sxp/loader.py
Comment (zain-sohail) [inline sed/loader/sxp/loader.py:10]: I think just doing a replace all of flash to sxp would be good

File: sed/loader/sxp/loader.py
Comment (zain-sohail) [inline sed/loader/sxp/loader.py:33]: can be removed

File: sed/loader/sxp/loader.py
Comment (zain-sohail) [inline sed/loader/sxp/loader.py:73]: I guess this would also be useful for flashloader

**Review by zain-sohail — CHANGES_REQUESTED** [review]
Generally I have seen the git blame so I got the idea where you made changes.
Is this going to be a permenant loader or it's just for your beamtime?
Because adapting flash loaders structure makes this confusing and likely unmaintainable. We should later definitely think of a solution to this.

File: sed/loader/flash/utils.py
Comment (rettigl) [reply sed/loader/flash/utils.py:1]: No, I moved the content (one function) into the general utils, as it is also used in the sxp loader now.

File: tests/data/loader/sxp/config.yaml
Comment (rettigl) [reply tests/data/loader/sxp/config.yaml:48]: This kind of defines a schema, but yes, reworking the mpes structure here is on my todo (not very high up though).

File: tests/data/loader/sxp/RAW-R0016-DA03-S00000.h5
Comment (rettigl) [reply tests/data/loader/sxp/RAW-R0016-DA03-S00000.h5:1]: Its a few trains from one of the runs. I clarified this with Markus Scholz.

File: tutorial/5_sxp_workflow.ipynb
Comment (rettigl) [reply tutorial/5_sxp_workflow.ipynb:1]: Not sue there will be a public example dataset soon, but I'd still like to merge this and make it usable easily.

File: sed/loader/sxp/metadata.py
Comment (rettigl) [reply sed/loader/sxp/metadata.py:1]: yes, I can install a dummy.

File: sed/loader/sxp/loader.py
Comment (rettigl) [reply sed/loader/sxp/loader.py:10]: I put this there on purpose ;) ...

Comment (rettigl) [reply sed/loader/sxp/loader.py:10]: this is the only ocurrence, btw

File: sed/loader/sxp/loader.py
Comment (rettigl) [reply sed/loader/sxp/loader.py:33]: yes

File: sed/loader/sxp/loader.py
Comment (rettigl) [reply sed/loader/sxp/loader.py:73]: There are a few things that seem inconsistent to me still. E.g. I think it would be good if you could auto-find the beamtime-dirs, and still user-define a parquet-dir. But that's something for another PR, I think.

**Comment by rettigl** [comment]
> Generally I have seen the git blame so I got the idea where you made changes. Is this going to be a permenant loader or it's just for your beamtime? Because adapting flash loaders structure makes this confusing and likely unmaintainable. We should later definitely think of a solution to this.

Well, In gerneral the requirements for FLASH and SXP are very similar, i.e. the train /micro/macrobunch structure etc., so it does not make sense to me to deviate too much from what we have. But I agree we need to find a way to maintain this. Either we outsource the common code into utils files, or build an intermedate FEL loader, where FLASH and SXP loader inherit from.

File: sed/loader/sxp/loader.py
Comment (zain-sohail) [reply sed/loader/sxp/loader.py:72]: I put this in #281

**Comment by zain-sohail** [comment]
> > Generally I have seen the git blame so I got the idea where you made changes. Is this going to be a permenant loader or it's just for your beamtime? Because adapting flash loaders structure makes this confusing and likely unmaintainable. We should later definitely think of a solution to this.
> 
> Well, In gerneral the requirements for FLASH and SXP are very similar, i.e. the train /micro/macrobunch structure etc., so it does not make sense to me to deviate too much from what we have. But I agree we need to find a way to maintain this. Either we outsource the common code into utils files, or build an intermedate FEL loader, where FLASH and SXP loader inherit from.

I think best solution is to define different classes for these different aspects and create the actual loaders to instantiate these classes

**Review by zain-sohail — APPROVED** [review]

**Pull request merged**
